Typos and Grammar fixes (#18)
tsl143/itsopensource: content/blog/get-started-with-web-wrokers/index.md

author: trishul
---

Javascript is single-threaded ie. all of the javascript code written is executed in a single thread. All the functions are executed sequentially. The next function will be executed once the previous one has finished its execution. This sometimes leads to unresponsive UI.
Consider this example, 
In part 1 when we click on button 1, the UI freezes for 2 seconds as the main thread is performing some CPU intensive operations. Until this execution is finished the button 2 is not clickable at all.
![part1](./part1.gif) The functionality of button 2 is independent of button 1 but still it's unusable until button 1's job is finished. This is a very common problem faced by javascript intensive web apps.

> The solution to this is **Web Workers** (*not Service workers*)

Reviewer: ```suggestion
The solution to this is **Web Workers** (*not Serviceworkers*)
```

author: trishul
---

Javascript is single-threaded ie. all of the javascript code written is executed in a single thread. All the functions are executed sequentially. The next function will be executed once the previous one has finished its execution. This sometimes leads to unresponsive UI.
Consider this example, 
In part 1 when we click on button 1, the UI freezes for 2 seconds as the main thread is performing some CPU intensive operations. Until this execution is finished the button 2 is not clickable at all.
![part1](./part1.gif) The functionality of button 2 is independent of button 1 but still it's unusable until button 1's job is finished. This is a very common problem faced by javascript intensive web apps.

> 

Reviewer: ```suggestion
```

self.postMessage(counter);
```
<br>

> This will be executed as soon as the worker is created we can adjust this to be called only when required via post message.

Reviewer: ```suggestion
This will be executed as soon as the worker is created we can adjust this to be called only when required via `postmessage`.
```

The output looks something like this
![part2](./part2.gif)

All the CPU intensive operations are happening in the worker while the UI is free and responsive. The complete code can be found [here](https://github.com/tsl143/itsopensource/tree/master/static/demos/webworkers).

> When it comes to loading time workers may not be making your web app load fast, but it ensures the main thread is free and the UI is not frozen. One of the rules I follow is; All UI updates should be done in the main thread and use workers for everything else.

Reviewer: ```suggestion
When it comes to loading time, workers may not be making your web app load fast, but it ensures the main thread is free and the UI is not frozen. One of the rules I follow is; All UI updates should be done in the main thread and use workers for everything else.
```
